### PR TITLE
docs - gcp pubsub examples

### PR DESCRIPTION
**Pull request by alexkarpitski**
Use cases for Pub/Sub `Topic`, `Subscription`, `Snapshot` resources.

### CODE REVIEW

File: docs/source/gcp/examples/pubsub-snapshots.rst [hunk 1]
```diff
@@ -0,0 +1,26 @@
+Pub/Sub - Find Snapshots Referring to Deleted Topics
+==================================================================
+
+In Cloud Pub/Sub, the snapshot feature allows users to capture the message acknowledgment state of a subscription to a topic. Once a snapshot is created, it retains all messages that were unacknowledged in the source subscription (at the time of the snapshot's creation). Custodian can check if any of Pub/Sub snapshots refer to deleted topics (GCP uses special ``_deleted-topic_`` name for them).
```
Comment (kapilt) [inline docs/source/gcp/examples/pubsub-snapshots.rst]: please note all snapshots auto delete after at *most* a week in the docs here.. which also decreases the utility of this policy imo beyond catering to edge case/long tail optimization..

File: docs/source/gcp/examples/pubsub-topics.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
+Pub/Sub - Find Topics which Violate Naming Convention
+=======================================================
```
Comment (kapilt) [inline docs/source/gcp/examples/pubsub-topics.rst]: please leave this example out, naming considerations on resources for an org don't need separate per resource examples.

Comment (alexkarpitski) [reply docs/source/gcp/examples/pubsub-topics.rst]: Done.

File: docs/source/gcp/examples/pubsub-snapshots.rst (hunk 1 above)
Comment (alexkarpitski) [reply docs/source/gcp/examples/pubsub-snapshots.rst]: That's right, however, per the updated description a week could still be too much in cases of regulations and someone would like to remove sensitive data sooner.

Comment (kapilt) [reply docs/source/gcp/examples/pubsub-snapshots.rst]: right.. this is a what if scenario thats deep down an edge case.. not a good doc example.

Comment (kapilt) [reply docs/source/gcp/examples/pubsub-snapshots.rst]: i cant think of a regulation scenario for it, and its pretty useless for most people

File: docs/source/gcp/examples/pubsub-snapshots.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
+Pub/Sub - Find Snapshots Referring to Deleted Topics
+====================================================
+
+In Cloud Pub/Sub, the snapshot feature allows users to capture the message acknowledgment state of a subscription to a topic. Once a snapshot is created, it retains all messages that were unacknowledged in the source subscription (at the time of the snapshot's creation).
+
+Some Pub/Sub snapshots may refer to deleted topics (GCP uses special ``_deleted-topic_`` name for them). Although such snapshots are deleted in a week upon creation, in order to enforce security Custodian can filter them earlier, thus decreasing time-to-live of unused sensitive data.
```
Comment (kapilt) [inline docs/source/gcp/examples/pubsub-snapshots.rst]: let's reword this entire paragraph, its not nesc a security thing, and we shouldn't assume as such in the docs when presenting generic corner case examples.

`All pubsub snapshots are deleted in a week or less of their creation. For some use cases it maybe useful to delete them earlier.`

Comment (aleonidex) [reply docs/source/gcp/examples/pubsub-snapshots.rst]: Thanks, I've generalized the title and description

**Pull request merged**